model/generate.go

```diff
@@ -102,65 +102,92 @@ func ParseProjectFromJSON(data []byte) (GeneratedProject, error) {
 
 // NewVersion adds the buildvariants, tasks, and functions
 // from a generated project config to a project, and returns the previous config string.
-func (g *GeneratedProject) NewVersion() (*Project, *Version, *task.Task, *projectMaps, string, error) {
+func (g *GeneratedProject) NewVersion() (*Project, *Version, *task.Task, *projectMaps, *parserProject, error) {
```

Comment:
public functions that return private types are... _a problem._

This is the kind of thing that should be ticketed if it's not meaningful to replace. 

Also, it's really too many return values, and I'm not sure what the task pointer is supposed to do?